src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilFluidBase.java

```diff
@@ -0,0 +1,120 @@
+package WayofTime.bloodmagic.item.sigil;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockLiquid;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fluids.IFluidBlock;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.wrappers.BlockLiquidWrapper;
+import net.minecraftforge.fluids.capability.wrappers.BlockWrapper;
+import net.minecraftforge.fluids.capability.wrappers.FluidBlockWrapper;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+public abstract class ItemSigilFluidBase extends ItemSigilBase {
+    //Class for sigils that interact with fluids, either creating or deleting them.
+    //Sigils still have to define their own onRightClick behavior, but the actual fluid-interacting code is largely limited to here.
+    public final FluidStack sigilFluid;
+    
+    public ItemSigilFluidBase(String name, int lpUsed, FluidStack fluid){
+        super(name, lpUsed);
+        sigilFluid = fluid;
+    }
+    public ItemSigilFluidBase(String name){
+        super(name);
+        sigilFluid = null;
+    }
+    
+    //The following are handler functions for fluids, all genericized.
+    //They're all based off of the Forge FluidUtil methods, but directly taking the sigilFluid constant instead of getting an argument.
+    
+    /* Gets a fluid handler for the targeted block and siding.
+     * Works for both tile entity liquid containers and fluid blocks.
+     * This one is literally identical to the FluidUtil method of the same signature.
+     */
+    protected IFluidHandler getFluidHandler(World world, BlockPos blockPos, EnumFacing side) {
+        IBlockState state = world.getBlockState(blockPos);
+        Block block = state.getBlock();
+        
+        if (block.hasTileEntity(state)) {
+            TileEntity tile = world.getTileEntity(blockPos);
+            if(tile != null && tile.hasCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, side))
```

Comment:
While it may make sense to normal people to do it like this, you're *apparently* not supposed to. This should be something like

```java
if (tile != null) {
    IFluidHandler fluidHandler = tile.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, side);
    if (fluidHandler != null)
        return fluidHandler;
}
```